File: src/uiMain/funcionalidades/Nomina.java
package uiMain.funcionalidades;

import java.time.temporal.ChronoField;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Scanner;
import java.util.SortedSet;
import java.util.TreeSet;

import gestorAplicacion.operacional.Cita;
import gestorAplicacion.operacional.Venta;
import gestorAplicacion.organizacional.Empleado;

public class Nomina {
////////////////////////////FUNCIONALIDAD NÓMINA ////////////////////////////////////////////////////////	

	public static void calcularNomina() {
		System.out.println("=============Pago nomina mes vencido=============");

		SortedSet<Integer> fechas = new TreeSet<Integer>();
		for (Cita cita : Cita.getCitas()) {
			fechas.add(cita.getFechaCita().get(ChronoField.MONTH_OF_YEAR));
		}
		for (Venta venta : Venta.getVentas()) {
			fechas.add(venta.getFechaVenta().get(ChronoField.MONTH_OF_YEAR));
		}

		System.out.println("");
		System.out.println("Seleccione el mes a pagar:");

		HashMap<Integer, String> meses = new HashMap<Integer, String>();
		meses.put(1, "Enero");
		meses.put(2, "Febrero");
		meses.put(3, "Marzo");
		meses.put(4, "Abril");
		meses.put(5, "Mayo");
		meses.put(6, "Junio");
		meses.put(7, "Julio");
		meses.put(8, "Agosto");
		meses.put(9, "Septiembre");
		meses.put(10, "Octubre");
		meses.put(11, "Noviembre");
		meses.put(12, "Diciembre");
		Iterator value = fechas.iterator();
		int i = 1;
		while (value.hasNext()) {
			System.out.println(i + " " + meses.get(value.next()));
			i++;
		}

		System.out.print("Opcionn: ");
		Scanner texto = new Scanner(System.in);
		int opcion = texto.nextInt();

		value = fechas.iterator();
		i = 1;
		int mes_escogido = 0;
		int valor = 0;
		while (value.hasNext()) {
			valor = (Integer) value.next();
			if (i == opcion) {
				mes_escogido = valor;
				break;
			}
			i++;
		}

//LOS EMPLEADOS
		System.out.println("");
		System.out.println("Seleccione el empleado a liquidar:");

		for (Empleado emp : Empleado.getEmpleados()) {
			System.out.println(emp);
		}

		System.out.print("Nombre: ");
		Scanner nombre = new Scanner(System.in);
		String empleado = nombre.next();

//Citas que atendió el empleado
		Empleado empleadoEscogido = null;
		for (Empleado emp : Empleado.getEmpleados()) {
			if (emp.getNombre().equals(empleado)) {
				empleadoEscogido = emp;

				break;
			}
		}
		ArrayList<Cita> citasMes = new ArrayList<Cita>();
		for (Cita cita : Cita.getCitas()) {
			if (cita.getFechaCita().get(ChronoField.MONTH_OF_YEAR) == mes_escogido) {

				if (cita.getEmpleado().getNombre().equals(empleado)) {
					citasMes.add(cita);

				}
			}
		}
//Productos que vendió el empleado

		ArrayList<Venta> ventasMes = new ArrayList<Venta>();
		for (Venta venta : Venta.getVentas()) {
			if (venta.getFechaVenta().get(ChronoField.MONTH_OF_YEAR) == mes_escogido) {
				if (venta.getEmpleadoComision().getNombre().equals(empleado)) {
					ventasMes.add(venta);
				}
			}
		}
		double totalComi = 0;
		for (Venta venta : ventasMes) {
			totalComi += (venta.getProductoVendido().getPrecioVenta() * Venta.porcentajeComision);
		}

		System.out.println(" ");
		System.out.println(" ");
		System.out.println(" ================================================================== ");
		System.out.println("                     Pago al empleado " + empleado);
		System.out.println(" ================================================================== ");
		System.out.println("  ");

		System.out.println("El empleado " + empleado + " atendio " + citasMes.size() + " citas en el mes");
		System.out.println("Tiene un sueldo base de:  " + empleadoEscogido.getSueldo());
		System.out.println(" ");

		System.out.println("Vendio: " + ventasMes.size() + " producto(s) en el mes, correspondientes a:");
		System.out.println(" ");
		for (Venta venta : ventasMes) {
			System.out.println("  --> " + venta.getProductoVendido().getNombreProducto() + " $ "
					+ venta.getProductoVendido().getPrecioVenta());
		}
		System.out.println(" ");
		System.out.println("Por lo que recibe una comision del 20% de cada producto, es decir: " + totalComi);

		System.out.println(" ");
		double totalTodo = empleadoEscogido.getSueldo() + totalComi;
		System.out.println("Confirmacion:");
		System.out.println("El sueldo total a pagar debe ser: " + totalTodo);

	}
}

File: Python/src/uiMain/Funcionalidades/Nomina.py
from organizacional.Empleado import Empleado
from operacional.Cita import Cita



class Nomina:
    
    #//////////////////////////FUNCIONALIDAD NÓMINA ////////////////////////////////////////////////////////	

    @staticmethod
    def calcularNomina():
        print("=============Pago nomina mes vencido=============")

        fechas = java.util.TreeSet()
        for cita in gestorAplicacion.operacional.Cita.getCitas():
            fechas.add(cita.getFechaCita().get(java.time.temporal.ChronoField.MONTH_OF_YEAR))
        for venta in gestorAplicacion.operacional.Venta.getVentas():
            fechas.add(venta.getFechaVenta().get(java.time.temporal.ChronoField.MONTH_OF_YEAR))

        print("")
        print("Seleccione el mes a pagar:")

        meses = {}
        meses[1] = "Enero"
        meses[2] = "Febrero"
        meses[3] = "Marzo"
        meses[4] = "Abril"
        meses[5] = "Mayo"
        meses[6] = "Junio"
        meses[7] = "Julio"
        meses[8] = "Agosto"
        meses[9] = "Septiembre"
        meses[10] = "Octubre"
        meses[11] = "Noviembre"
        meses[12] = "Diciembre"
        value = fechas.iterator()
        i = 1
        while value.hasNext():
            print(str(i) + " " + meses[value.next()])
            i += 1

        print("Opcionn: ", end = '')
        texto = java.util.Scanner(System.in)
        opcion = texto.nextInt()

        value = fechas.iterator()
        i = 1
        mes_escogido = 0
        valor = 0
        while value.hasNext():
            valor = int(value.next())
            if i == opcion:
                mes_escogido = valor
                break
            i += 1

        #LOS EMPLEADOS
        print("")
        print("Seleccione el empleado a liquidar:")

        for emp in Empleado.getEmpleados():
            print(emp)

        print("Nombre: ", end = '')
        nombre = java.util.Scanner(System.in)
        empleado = nombre.next()

        #Citas que atendió el empleado
        empleadoEscogido = None
        for emp in gestorAplicacion.organizacional.Empleado.getEmpleados():
            if emp.getNombre() is empleado:
                empleadoEscogido = emp

                break
        citasMes = []
        for cita in gestorAplicacion.operacional.Cita.getCitas():
            if cita.getFechaCita().get(java.time.temporal.ChronoField.MONTH_OF_YEAR) == mes_escogido:

                if cita.getEmpleado().getNombre() is empleado:
                    citasMes.append(cita)

        #Productos que vendió el empleado

        ventasMes = []
        for venta in gestorAplicacion.operacional.Venta.getVentas():
            if venta.getFechaVenta().get(java.time.temporal.ChronoField.MONTH_OF_YEAR) == mes_escogido:
                if venta.getEmpleadoComision().getNombre() is empleado:
                    ventasMes.append(venta)
        totalComi = 0
        for venta in ventasMes:
            totalComi += (venta.getProductoVendido().getPrecioVenta() * gestorAplicacion.operacional.Venta.porcentajeComision)

        print(" ")
        print(" ")
        print(" ================================================================== ")
        print("                     Pago al empleado " + empleado)
        print(" ================================================================== ")
        print("  ")

        print("El empleado " + empleado + " atendio " + str(len(citasMes)) + " citas en el mes")
        print("Tiene un sueldo base de:  " + empleadoEscogido.getSueldo())
        print(" ")

        print("Vendio: " + str(len(ventasMes)) + " producto(s) en el mes, correspondientes a:")
        print(" ")
        for venta in ventasMes:
            print("  --> " + venta.getProductoVendido().getNombreProducto() + " $ " + venta.getProductoVendido().getPrecioVenta())
        print(" ")
        print("Por lo que recibe una comision del 20% de cada producto, es decir: " + str(totalComi))

        print(" ")
        totalTodo = empleadoEscogido.getSueldo() + totalComi
        print("Confirmacion:")
        print("El sueldo total a pagar debe ser: " + str(totalTodo))

